plume-models/src/search/query.rs

```diff
@@ -259,25 +241,28 @@ impl PlumeQuery {
     }
 
     // recursive parser for query string
+    // allow this clippy lint for now, until someone figures out how to
+    // refactor this better.
+    #[allow(clippy::wrong_self_convention)]
```

Comment:
seeing `parse_query`, I think this function could be moved into it, and then we can call `parse_query` instead of `from_str_req`